src/main/java/io/jenkins/plugins/jfrog/JfStep.java

```diff
@@ -100,6 +96,46 @@ public void perform(@NonNull Run<?, ?> run, @NonNull FilePath workspace, @NonNul
         }
     }
 
+    /**
+     * Configure all JFrog relevant environment variables and all servers (if they haven't been configured yet).
+     *
+     * @param run             running as part of a specific build
+     * @param env             environment variables applicable to this step
+     * @param launcher        a way to start processes
+     * @param listener        a place to send output
+     * @param workspace       a workspace to use for any file operations
+     * @param jfrogBinaryPath path to jfrog cli binary on the filesystem
+     * @param isWindows       is Windows the applicable OS
+     * @return launcher applicable to this step.
+     * @throws InterruptedException if the step is interrupted
+     * @throws IOException          in case of any I/O error, or we failed to run the 'jf' command
+     */
+    public Launcher.ProcStarter setupJFrogEnvironment(Run<?, ?> run, EnvVars env, Launcher launcher, TaskListener listener, FilePath workspace, String jfrogBinaryPath, boolean isWindows) throws IOException, InterruptedException {
+        // Set relevant environment variables.
+        if (env.get(JFROG_CLI_BUILD_NAME) == null) {
+            // Setting Jenkins job name as the default build-info name.
+            env.put(JFROG_CLI_BUILD_NAME, env.get("JOB_NAME"));
+        }
+        if (env.get(JFROG_CLI_BUILD_NUMBER) == null) {
+            // Setting Jenkins build number as the default build-info number.
+            env.put(JFROG_CLI_BUILD_NUMBER, env.get("BUILD_NUMBER"));
+        }
+        if (env.get(JFROG_CLI_BUILD_URL) == null) {
+            // Setting the specific build URL.
+            env.put(JFROG_CLI_BUILD_URL, env.get("BUILD_URL"));
+        }
```

Comment:
We can remove code duplication by doing the following:
```suggestion
        // Setting Jenkins job name as the default build-info name.
        env.putIfAbsent(JFROG_CLI_BUILD_NAME, env.get("JOB_NAME"));
        // Setting Jenkins build number as the default build-info number.
        env.putIfAbsent(JFROG_CLI_BUILD_NUMBER, env.get("BUILD_NUMBER"));
        // Setting the specific build URL.
        env.putIfAbsent(JFROG_CLI_BUILD_URL, env.get("BUILD_URL"));
```